File: src/spice2x/api/resources/dart/spiceapi/src/wrappers/keypads.dart
part of spiceapi;

Future<void> keypadsWrite(Connection con, int unit, String input) {
  var req = Request("keypads", "write");
  req.addParam(unit);
  req.addParam(input);
  return con.request(req);
}

Future<void> keypadsSet(Connection con, int unit, String buttons) {
  var req = Request("keypads", "set");
  req.addParam(unit);
  for (int i = 0; i < buttons.length; i++)
    req.addParam(buttons[i]);
  return con.request(req);
}

Future<String> keypadsGet(Connection con, int unit) {
  var req = Request("keypads", "get");
  req.addParam(unit);
  return con.request(req).then((res) {
    String buttons = "";
    for (var obj in res.getData()) {
      buttons += obj;
    }
    return buttons;
  });
}

File: src/spice2x/api/resources/python/spiceapi/keypads.py
from .connection import Connection
from .request import Request


def keypads_write(con: Connection, keypad: int, input_values: str):
    req = Request("keypads", "write")
    req.add_param(keypad)
    req.add_param(input_values)
    con.request(req)


def keypads_set(con: Connection, keypad: int, input_values: str):
    req = Request("keypads", "set")
    req.add_param(keypad)
    for value in input_values:
        req.add_param(value)
    con.request(req)


def keypads_get(con: Connection, keypad: int):
    req = Request("keypads", "get")
    req.add_param(keypad)
    res = con.request(req)
    return res.get_data()
